src/main/java/se/sundsvall/billingdatacollector/api/CollectorResource.java

```diff
@@ -0,0 +1,108 @@
+
+import se.sundsvall.billingdatacollector.service.CollectorService;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotEmpty;
+
+@RestController
+@Validated
+@Tag(name = "Billing Data Collector", description = "Billing Data Collector resources")
+@RequestMapping(path = "/")
+@ApiResponse(
+	responseCode = "400",
+	description = "Bad Request",
+	content = @Content(schema = @Schema(oneOf = {Problem.class, ConstraintViolationProblem.class})))
+@ApiResponse(
+	responseCode = "500",
+	description = "Internal Server Error",
+	content = @Content(schema = @Schema(implementation = Problem.class)))
+@ApiResponse(
+	responseCode = "502",
+	description = "Bad Gateway",
+	content = @Content(schema = @Schema(implementation = Problem.class)))
+public class CollectorResource {
+
+	private final CollectorService collectorService;
+
+	public CollectorResource(CollectorService collectorService) {
+		this.collectorService = collectorService;
+	}
+
+	@Operation(
+		summary = "Trigger billing for a flowInstanceId",
+		responses = {
+			@ApiResponse(
+				responseCode = "202",
+				description = "Accepted",
+				useReturnTypeSchema = true)
+		}
+	)
+	@PostMapping(path = "/trigger/{flowInstanceId}", produces = { APPLICATION_JSON_VALUE, APPLICATION_PROBLEM_JSON_VALUE })
+	ResponseEntity<Void> triggerBilling(
+		@Parameter(name = "flowInstanceId", description = "flowInstanceId to trigger billing for", example = "123")
+		@NotEmpty @PathVariable("flowInstanceId") final String flowInstanceId) {
+
+		collectorService.trigger(flowInstanceId);
+
+		return ResponseEntity.accepted().build();
+	}
+
+	@Operation(
+		summary = "Trigger billing for all flowInstancesId:s between two specific dates",
+		responses = {
+			@ApiResponse(
+				responseCode = "202",
+				description = "Accepted",
+				useReturnTypeSchema = true)
+		}
+	)
+	@PostMapping(path = "/trigger", produces = { APPLICATION_JSON_VALUE, APPLICATION_PROBLEM_JSON_VALUE })
+	ResponseEntity<Void> triggerBilling(
+		@RequestParam(name = "startDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
+		@Valid @Parameter(example = "2024-01-01") final LocalDate startDate,
+
+		@RequestParam(name = "endDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
+		@Valid @Parameter(example = "2024-02-01") final LocalDate endDate) {
+
+		validateDateRange(startDate, endDate);
+
+		collectorService.triggerBetweenDates(startDate, endDate);
+
+		return ResponseEntity.accepted().build();
+	}
+
+	//Validate that the end date is after, or equal to, the start date
+	private void validateDateRange(LocalDate startDate, LocalDate endDate) {
+		if(startDate.isAfter(endDate)) {
```

Comment:
This should be verified with regard to the OpenE "API", but I suspect that we will have to support passing the same date as from and to, and therefore, this should be something like `startDate.isEqualOrAfter(endDate)` (that doesn't exist out-of-the-box...)

Reply:
The logic is correct though, but I'll rewrite it to make it readable